waiter/src/waiter/interstitial.clj

```diff
@@ -94,29 +94,25 @@
    In particular, it resolves all promises for services which have healthy instances.
    It also removes from the interstitial state any resolved promises for services which are no longer available."
   [interstitial-state-atom service-id->service-description current-available-service-ids scheduler-messages]
-  (let [available-service-ids (->> scheduler-messages
-                                   (some (fn [[message-type message-data]]
-                                           (when (= :update-available-services message-type)
-                                             (:available-service-ids message-data))))
-                                   set)
-        healthy-instance-service-ids (->> scheduler-messages
-                                          (keep (fn [[message-type message-data]]
-                                                  (when (and (= :update-service-instances message-type)
-                                                             (seq (:healthy-instances message-data)))
-                                                    (:service-id message-data)))))
+  (let [summary-data (-> (fn [[message-type message-data]]
```

Comment:
Remove this threading macro

Reply:
Done, ended up improving the snippet quite a bit.